etl/scripts-ccao-data-warehouse-us-east-1/spatial/spatial-ccao-corner.R

```diff
@@ -0,0 +1,236 @@
+    # order to fill any tiny gaps between abutting parcels
+    town_parcels_buf <- parcels %>%
+      filter(
+        between(lon, town_bbox$xmin, town_bbox$xmax) &
+          between(lat, town_bbox$ymin, town_bbox$ymax)
+      ) %>%
+      st_buffer(dist = units::set_units(2, "m"))
+
+    # Fetch the OSM street network for the township, removing any OSM way types
+    # that are not main roads
+    osm_streets <- opq(bbox = town_bbox) %>%
+      add_osm_feature(key = "highway") %>%
+      osmdata_sf() %>%
+      .$osm_lines %>%
+      filter(
+        !highway %in% c(
+          "bridleway", "construction", "corridor", "cycleway", "elevator",
+          "service", "services", "steps", "platform", "motorway",
+          "motorway_link", "pedestrian", "track", "path", "footway", "alley"
+        )
+      ) %>%
+      st_transform(3435)
+
+    # Step 1: Find the minimum rectangle that bounds the parcel, then use that
+    # rectangle to determine the parcel's orientation and length. These values
+    # are used in the next step to draw a cross on the parcel
+    town_mrr <- town_parcels %>%
+      st_minimum_rotated_rectangle() %>%
+      select(lon, lat, geometry = geometry_3435) %>%
+      mutate(id = row_number()) %>%
+      st_transform(4326) %>%
+      st_cast("POINT") %>%
+      mutate(
+        length = st_distance(geometry, lead(geometry), by_element = TRUE) +
+          units::set_units(40, "m"),
+        bearing = c(
+          lwgeom::st_geod_azimuth(geometry),
+          units::set_units(NA, "radians")
+        ),
+        bearing = units::set_units(bearing, "degrees")
+      ) %>%
+      filter(row_number() %% 5 != 0) %>%
+      # To suppress the warning about repeating constant attributes when casting
+      # geometries to POINT
+      suppressWarnings()
+
+    # Step 2: Draw a cross that cuts the parcel in four and extends out slightly
+    # beyond the minimum rectangle of the parcel. We will look for parcels and
+    # streets that intersect this cross in order to determine neighbors. Use the
+    # bearing and length calculated in the previous step to draw the cross
+    town_parcels_centroid <- town_mrr %>%
+      st_drop_geometry() %>%
+      st_as_sf(coords = c("lon", "lat"), crs = 4326) %>%
+      st_coordinates()
+
+    cross <- geosphere::destPoint(
+        p = town_parcels_centroid,
+        b = town_mrr$bearing,
+        d = town_mrr$length
+      ) %>%
+      cbind(town_parcels_centroid) %>%
+      as.data.frame()
+
+    # Draw the lines that connect the centroids to the endpoints of the cross
+    draw_line <- function(r) st_linestring(t(matrix(unlist(r), 2, 2)))
+    cross$geometry <- st_sfc(sapply(
+      seq_len(nrow(cross)),
+      function(i) draw_line(cross[i, ]),
+      simplify = FALSE
+    ))
+
+    # Compute the length and aspect ratio of the cross. The aspect ratio will be
+    # useful for filtering out parcels that are dramatically longer than wide
```

Comment:
**[Question, non-blocking]** Am I right that we're no longer computing or using aspect ratio here?

Reply:
Correct! Removed in 6c7a4c4f65b1c729a20575d19633d9c1bc00adaa.